components/usage/pkg/db/workspace_instance_usage_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package db_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/gitpod-io/gitpod/usage/pkg/db"
+	"github.com/gitpod-io/gitpod/usage/pkg/db/dbtest"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCanCreateUsageRecords(t *testing.T) {
+	conn := dbtest.ConnectForTests(t)
+	teamID := uuid.New().String()
+	teamAttributionID := db.NewTeamAttributionID(teamID)
+
+	scenarios := []struct {
+		Name     string
+		Report   map[db.AttributionID][]db.WorkspaceInstanceForUsage
+		Expected map[string]int64
+	}{
+		{
+			Name: "One workspace instance",
+			Report: map[db.AttributionID][]db.WorkspaceInstanceForUsage{
+				teamAttributionID: {
+					db.WorkspaceInstanceForUsage{
+						// has 1 day and 23 hours of usage
+						UsageAttributionID: teamAttributionID,
+						WorkspaceClass:     "default",
+						CreationTime:       db.NewVarcharTime(time.Date(2022, 05, 30, 00, 00, 00, 00, time.UTC)),
+						StoppedTime:        db.NewVarcharTime(time.Date(2022, 06, 1, 1, 0, 0, 0, time.UTC)),
+					},
+				},
+			},
+		},
+	}
+
+	ctx := context.Background()
+	for _, scenario := range scenarios {
+		t.Run(scenario.Name, func(t *testing.T) {
+			require.NoError(t, db.CreateUsageRecords(ctx, conn, scenario.Report))
+		})
+	}
+}
+
+func TestCanHandleMultipleReportsWithDuplicateEntries(t *testing.T) {
+	conn := dbtest.ConnectForTests(t)
```

Comment:
I'd recommend moving the `conn` into each test scenario, to ensure that after each test the data is deleted before the next test run

Reply:
done